### PR TITLE
Extend user permission tests to cover all types

### PR DESCRIPTION
**Pull request by mgencur**
https://issues.redhat.com/browse/SRVCOM-854

### CODE REVIEW

**Comment by mgencur** [comment]
Rebased. Let's see what happens. The failures looked unrelated.

**Comment by mgencur** [comment]
The error in the upgrade tests is: `prober.go:103: actual unavailable period 1m1.032662026s is over down time limit of 1m0s` 
We've seen it a few times and we'll need to decide if the 1minute limit is OK or should be increased with the latest migration to the new Kafka Broker components.
Anyway, this failure has nothing to do with this PR. I think it can be merged.

File: test/extensione2e/kafka/user_permissions_test.go [hunk 1]
```diff
@@ -49,31 +52,62 @@ func TestKafkaUserPermissions(t *testing.T) {
...
+		Spec: kafkachannelv1beta1.KafkaChannelSpec{
+			NumPartitions:     1,
+			ReplicationFactor: 1,
+		},
+	}
+
+	kafkaSink := &kafkasinksv1alpha1.KafkaSink{
+		Spec: kafkasinksv1alpha1.KafkaSinkSpec{
+			Topic:             "my-topic",
+			NumPartitions:     pointer.Int32Ptr(10),
```
Comment (skonto) [inline test/extensione2e/kafka/user_permissions_test.go:65]: How about using `ptr.Int32(10)` and import "knative.dev/pkg/ptr" instead?
Same for ` ptr.String(kafkasinksv1alpha1.ModeStructured)`.

Comment (mgencur) [reply test/extensione2e/kafka/user_permissions_test.go:65]: I could do that. The two functions have absolutely same result. Just for my understanding - is there any benefit?

File: go.mod [hunk 2]
```diff
@@ -159,7 +159,7 @@ require (
 	k8s.io/gengo v0.0.0-20211129171323-c02415ce4185 // indirect
 	k8s.io/klog/v2 v2.40.1 // indirect
 	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65 // indirect
-	k8s.io/utils v0.0.0-20211208161948-7d6a63dca704 // indirect
+	k8s.io/utils v0.0.0-20211208161948-7d6a63dca704
```
Comment (skonto) [inline go.mod:162]: Update this too?

File: test/extensione2e/kafka/user_permissions_test.go (hunk 1 above)
Comment (skonto) [reply test/extensione2e/kafka/user_permissions_test.go]: Avoid introducing new deps.

File: go.mod (hunk 2 above)
Comment (mgencur) [reply go.mod]: Yeah. Now I understand. Thanks. Pushed.

**Comment by mgencur** [comment]
/retest
Infra issue.

**Pull request merged**